### DISCUSSION TITLE
When the callback NotificationListener.onSubscriptionTransferFailed occurs, how do we restore the subscription?

### DISCUSSION THREAD

**Discussion by cityWalker312** [eclipse-milo/milo#1080] [General]
**Describe the bug**
Here is the code segment。
```java
subscription.addNotificationListener(new SubNotifierListener());
```
```java
class SubNotifierListener implements UaSubscription.NotificationListener {
    @Override
    public void onSubscriptionTransferFailed(UaSubscription subscription, StatusCode statusCode) {
        if (subscription == null || statusCode == null) {
            return;
        }
        printWarningMsg(subscription, "onSubscriptionTransferFailed", statusCode.getValue());
    }
}
```
**Expected behavior**
Can you give best practice advice for resuming subscriptions？

**Logs and Packet Captures**

Here is the log segment

---
onSubscriptionTransferFailed error code=0x80280000, warn msg: subId=85, items size=1

**Additional context**
Milo client sdk version is 0.6.3
Milo server sdk version is 0.6.3

**Comment by kevinherron** [comment]
Presumably your application has some bookkeeping about what the user is interested in. 

You created that subscription and added items to it once already. You know how to do it. Now you've been told that you need to do it again.

Create a new subscription. Create new monitored items belonging to that subscription.

**Reply by cityWalker312** [reply]
> Presumably your application has some bookkeeping about what the user is interested in.
> 
> You created that subscription and added items to it once already. You know how to do it. Now you've been told that you need to do it again.
> 
> Create a new subscription. Create new monitored items belonging to that subscription.

Your suggestion is that we simply recreate the subscription when NotificationListener.onSubscriptionTransferFailed occurs. 
Do old subscriptions automatically release resources on the client side and server side of the stack?

**Reply by kevinherron** [reply]
They don't exist on the server any more, else we would have been able to resume the session or transfer them.

In the client they only exist in memory. Delete any references to the subscription object you may be holding and let it be garbage collected. The subscription manager has already stopped bookkeeping it at this point.

**Reply by cityWalker312** [reply]
Thank you for your reply，:)
